### ISSUE TITLE
Shewit : Workflows, 2 weeks

### ISSUE THREAD

**Issue by Shewitttt** [lab-brussels-1/home#197]
> [my fork of module repository]()

# Learning Objectives
- Setup: GitHub
  - [x] created separate issues for separate .md files
  - [x] assigned myself in all issues
  - [x] moved all issue to the project board
- Setup: local
  - [x] pulled the most recent code from `main`/`master`
  - [x] created a new branch for my introduction
- Writing the content:
  - [ ] added my intro file in `/people` according to group conventions
  - [x] linked to my file from `/README.md` according to group conventions
- Before pushing my branch:
  - [x] formatted the code
  - [x] fixed all spelling mistakes and added new words to the dictionary
  - [x] fixed all linting mistakes
- After pushing my branch:
  - [x] created a new PR linked to this issue
  - [x] make sure all of the checks passed
  - [x] assigned someone to review and merge my PR
  ### 🥚 [1. Local Development With Git](./1-local-development-with-git)

Practice using Git to save and organize your development process. You will learn
how you can use Git to go back to previous versions of your project, and to work
on different changes in parallel.

- 🥚 **Git**: Using the CLI you can ...
  - [x] Initialize a new git repository with `git init`
  - [x] stage changes using `git add <path>`
  - [x] commit changes using `git commit -m <message>`
  - [x] display your repository's git history using `git log`
  - [x] create a new branch using `git branch <branch-name>`
  - [x] check out a branch using `git checkout <branch-name>`
  - [x] create a new branch and check it out using
        `git checkout -b <branch-name>`
  - [x] merge changes from one branch to another using `git merge <branch-name>`
  - [ ] return to a previous version of your project with `git log` and
        `git checkout <commit-hash>`
  - [ ] stash and retrieve uncommitted changes with `git stash` and `git pop`
- 🥚 **`.gitignore`**: You can use a `.gitignore` file to describe which files
  you don't want included in your git history.
- 🥚 **VSCode**: You can ...
  - [ ] use the _Git Graph_ extension to to visualize your repo's commit history
  - [ ] use the _Git Lens_ extension to investigate your repository's commit
        history
  - [ ] explain how the file tree in VSCode can show you which files have
        uncommitted changes
- [ ] 🥚 **Atomic Commits**: You can save your development progress using small
      commits with clear and helpful message.
- [x] 🐣 **Feature Branches**: You can organize your development process using
      branches. You can create a new branch for each part of your project and
      merge those changes to `main` when they are finished.

### 🥚 [1. Local/Remote Development](./2-local-remote-development)

Learn how you can connect your local Git repositories with a GitHub repository
to add more structure to your development process and to share your projects.

- 🥚 **Licenses**: You can ...
  - [ ] explain why it's important to include a license for your code on GitHub
  - [ ] choose a license for your projects that matches how you want others to
        use your code
- [x] 🥚 **GitHub SSH Key**: You can connect your computer to your GitHub
      account using an SSH key, clone using the SSH link, and push/pull using
      your SSH connection.
- 🥚 **GitHub Repositories**: You can ...
  - [x] create new repository on GitHub
  - [x] write a description for your repository
  - [x] turn on GitHub Pages
  - [x] configure your repository to block pushing to `main`
  - [x] configure your repository to block merging to `main` until Continuous
        Integration (CI) checks have passed
- 🥚 **Git Remote/Local Connection**: You can ...
  - [x] clone a remote repository to your computer
  - [x] initialize a new repository locally and connect it to an empty remote
        repository
  - [x] `push` and `pull` changes between remote & local branches
- [x] 🥚 **Pull Requests**: You can create a pull request between two branches
      in your repository and merge changes without causing any conflicts.
- [x] 🥚 **PR templates**: You can use a PR template to add a checklist to all
      of your PRs so you are sure the code is great before merging to `main`.
- [x] 🥚 **Continuous Integration**: You can use GitHub Actions to check your
      code's quality before merging a pull request to the `main` branch.
- [x] 🐣 **Local/Remote Branching Workflow**: You can use a branching workflow
      that keeps mistakes away from the `main` branch and prevents conflicts
      from happening in GitHub. For each contribution to the project you can ...
  1. Check out a new local branch and write your code
  2. Check out `main` on your local machine
  3. Pull changes from remote `main` to local `main`
  4. Merge changes locally from `main` to your new branch
     - Fix any conflicts on your new branch before pushing!
  5. Format and lint your code
  6. Push your new local branch to your remote repository
  7. Open a Pull Request from the new branch to `main`
     - Go through the PR's checklist to make sure everything is correct
     - Make sure all CI checks pass!
  8. Merge your new remote branch to `main`
     - You can delete the branch after it's merged if you want to

### 🐣 [3. Remote Collaboration](./3-remote-collaboration)

Learn how to collaborate with a group on a single project hosted in a GitHub
repository. Practice using GitHub's project management features to organize your
group's tasks and to double-check your project's code quality.

- [ ] 🥚 **Contributor Guidelines**: Your group can define contributor
      guidelines that describes how everyone can add their work to the project.
- [ ] 🥚 **Code of Conduct**: Your group can write a code of conduct for your
      project that describes how everyone should communicate and what behavior
      is considered unproductive.
- [x] 🥚 **Repository Contributors**: You can add group members as collaborators
      in a repository.
- [x] 🥚 **Issue Templates**: Your group can use issue templates to make sure
      all issues are complete and relevant to the project. in your repository
      and merge changes without causing any conflicts.
- [ ] 🥚 **Code Review**: You can use the PR Template and CI checks to review
      another group member's code before merging it to `main`.
- 🐣 **GitHub Project Management**: You can ...
  - [x] use issues to define tasks in a group project. Each issue should have a
        helpful title, complete description, and helpful labels.
  - [x] use a Project Board to organize a project's issues
  - [x] claim issues and track your progress with the project board
  - [x] link a PR to your claimed issue
  - [x] assign someone to review your PR
  - [ ] use GitHub code review features to discuss your code with your reviewer

**Comment by danielhalasz** [comment]
@Shewitttt no need to create a new issue, just add the comment to your previous one..see my post on Discord..I am closing this now..thx
